tests/react/transition.test.tsx

```diff
@@ -51,12 +52,12 @@ describeWithUseTransition('useTransition', () => {
     )
 
     resolve()
-    await findByText('delayed: 0')
+    await waitFor(() => expect(getByText('delayed: 0')).toBeTruthy())
 
-    fireEvent.click(getByText('button'))
-    await waitFor(() => {
+    await userEvent.click(getByText('button'))
+    await waitFor(async () => {
       resolve()
-      getByText('delayed: 1')
+      return expect(await findByText('delayed: 1')).toBeTruthy()
     })
```

Comment:
Another experimental suggestion. Can this block be like this?
```ts
  resolve()
  await waiFor(() => getByText('delayed: 1')).toBeTruthy())
```

Reply:
We could only have `matchers` on `expect` so this wouldn't work 😟 `getByText` returns an `HTMLElement` instead of a `Matcher`

Reply:
oops, that's my typo.
```js
  resolve()
  await waiFor(() => expect(getByText('delayed: 1')).toBeTruthy())
```

Reply:
This wouldn't work either because `waitFor` returns a promise

Reply:
Yeah, no luck with this either! But wrapping `resolve` around `act` seems to make a difference 🤔

Reply:
Thanks, I think it's better. (I may change my mind in the future. 😝 )